pkg/cmd/grafana-cli/commands/install_command.go

```diff
@@ -157,32 +157,35 @@ func latestSupportedVersion(plugin *m.Plugin) *m.Version {
 
 // SelectVersion returns latest version if none is specified or the specified version. If the version string is not
 // matched to existing version it errors out. It also errors out if version that is matched is not available for current
-// os and platform.
+// os and platform. It expects plugin.Versions to be sorted so the newest version is first.
 func SelectVersion(plugin *m.Plugin, version string) (*m.Version, error) {
```

Comment:
Refactored the code a bit to fix the bugs and make it clearer. There were some assumptions (mainly that API will filter versions that should not work in current architecture) which made the code work but with the refactor it should work correctly even without those assumptions. Though the assumption that versions are sorted is still required as we do not strictly rely on versioning scheme (and thus meaningful comparison of version strings).